Source/Scene/Primitive.js

```diff
@@ -798,6 +811,7 @@ define([
                     }, function(error) {
                         that._error = error;
                         that._state = PrimitiveState.FAILED;
+                        this._readyPromise.resolve(this);
```

Comment:
That's a good idea, I ran into a similar problem recently with the zoom stuff that caused be to insert an extra render in Viewer code; I think using `afterRender` there will fix that too. Thanks for the heads up.